File: domoticz_lcd4x20.php
#!/usr/bin/php
<?php

	$device=("/dev/ttyAMA0");
	$lcd=fopen($device,"wb+");
	fputs($lcd,chr(20));
	fputs($lcd,chr(150));
	fputs($lcd,chr(12));
//	fputs($lcd,'test');
//	fclose($lcd);

while (1) {
	$json_string = file_get_contents("http://DOMOTICZ_URL:DOMOTICZ_PORT/json.htm?type=devices&used=true&filter=all&favorite=1");
	$parsed_json = json_decode($json_string, true);
	$link = "/var/tmp/data.json";
	$data = fopen ($link, "w+");
	fwrite ($data, print_R($parsed_json, TRUE));
	fclose ($data);
	echo date('h:i:s')."\n";

	// first screen
	$parsed_json = $parsed_json['result'][0];
	$temp_ext = $parsed_json['Temp'];
	echo "Ext temp ".$temp_ext." C"."\n";
	fputs($lcd,"Ext temp ".$temp_ext." C"."\n");
	$parsed_json = json_decode($json_string, true);
	$parsed_json = $parsed_json['result'][0];
	$hum = $parsed_json['Humidity'];
	echo "Humidity ".$hum." %"."\n";
	fputs($lcd,"Ext hum  ".$hum." %"."\n");
	$parsed_json = json_decode($json_string, true);
	$parsed_json = $parsed_json['result'][2];
	$bar = $parsed_json['Barometer'];
	echo "Barometer ".$bar." mBar"."\n";
	fputs($lcd,"Barom    ".$bar." mb"."\n");

	echo  date("H:i d-m-Y")."\n";
	fputs($lcd,date("H:i    d-m-Y"."\n"));
	sleep(10);

	// second screen
	fputs($lcd,chr(12));
	$parsed_json = json_decode($json_string, true);
	$parsed_json = $parsed_json['result'][2];
	$temp_int = $parsed_json['Temp'];
	echo "Int temp ".$temp_int." C"."\n";
	fputs($lcd,"Int temp ".$temp_int." C"."\n");
	$parsed_json = json_decode($json_string, true);
	$parsed_json = $parsed_json['result'][1];
	$rain = $parsed_json['Rain'];
	echo "Rain     ".$rain." mm"."\n";
	fputs($lcd,"Rain     ".$rain." mm"."\n\n");
	echo  date("H:i d-m-Y")."\n";
	fputs($lcd,date("H:i    d-m-Y"."\n"));
	sleep(10);

	fputs($lcd,chr(12));
}
?>

File: domoticz_lcd4x20.py
#!/usr/bin/python

import urllib
import json
import time

#read domoticz data from my_station

while 1:
    url="http://raspi-meteo.local:8080/json.htm?type=devices&used=true&filter=all&favorite=1"

    data = json.load(urllib.urlopen(url))

    #print(json.dumps(data, indent=4, sort_keys=True))

    parsed_json = data['result'][1]
    temp= parsed_json['Temp']

    parsed_json = data['result'][1]
    hum = parsed_json['Humidity']

    parsed_json = data['result'][2]
    dir = parsed_json['Direction']

    parsed_json = data['result'][2]
    speed = parsed_json['Speed']

    parsed_json = data['result'][0]
    rain = parsed_json['Rain']

    print 'Temp: ',temp,'Hum: ',hum,'Wind dir: ',dir,'Wind speed: ',speed,'Rain: ',rain

    time.sleep(5)
